ts/src/test/Exchange/base/test.borrowInterest.ts

```diff
@@ -12,14 +12,14 @@ function testBorrowInterest (exchange, method, entry, requestedCode, requestedSy
         'timestamp': 1638230400000,
         'datetime': '2021-11-30T00:00:00.000Z',
     };
-    const emptyNotAllowedFor = [ 'currency', 'interest', 'interestRate', 'amountBorrowed', 'timestamp' ];
-    testSharedMethods.assertStructure (exchange, method, entry, format, emptyNotAllowedFor);
-    testSharedMethods.assertTimestamp (exchange, method, entry);
-    testSharedMethods.assertCurrencyCode (exchange, method, entry, entry['currency'], requestedCode);
-    testSharedMethods.assertSymbol (exchange, method, entry, entry['account'], requestedSymbol);
-    testSharedMethods.assertGreater (exchange, method, entry, 'interest', '0');
-    testSharedMethods.assertGreater (exchange, method, entry, 'interestRate', '0');
-    testSharedMethods.assertGreater (exchange, method, entry, 'amountBorrowed', '0');
+    const emptyAllowedFor = [ 'account' ];
+    testSharedMethods.assertStructure (exchange, skippedProperties, method, entry, format, emptyAllowedFor);
+    testSharedMethods.assertTimestamp (exchange, skippedProperties, method, entry);
+    testSharedMethods.assertCurrencyCode (exchange, skippedProperties, method, entry, entry['currency'], requestedCode);
+    testSharedMethods.assertSymbol (exchange, skippedProperties, method, entry, entry['account'], requestedSymbol);
```

Comment:
I'm unsure if we should provide the `skippedProperties` array to these assert methods that only test one value, what if instead of doing this:

```
testSharedMethods.assertTimestamp (exchange, 'timestamp' in skippedProperties, method, entry);
```
or a bit more verbose (but also more clear)

```
if (!'timestamp' in skippedProperties) {
    testSharedMethods.assertTimestamp (exchange, skippedProperties, method, entry);
}
```

what do you think?

Reply:
will it hurt if we pass whole `skipProperties` for now and don't change it in this PR?

Reply:
I have to continue works on test files and might touch this in near future, but for now, the best way was to pass whole `skipProps` object

Reply:
but yeah, in some methods, they might not be used at all, or just 1 field, but I dont like to be too much detailed to divide methods where "just one property" is used from "skipProps" or multiple, and use `if (!'timestamp' in skippedProperties)` for some methods, and don't use for other methods. I prefer to be uniform approach, just pass skipProps and in methods decide whether use skipProps at all, or use just 1 field, or use multiple fields. I believe from tests we should pass it as whole, and they should be agnostic of implementations needs (as we have to update some things yet in implementations)

Reply:
> 

@ttodua we can keep it as is for now and reevaluate later, but for instance providing a list with `['last', 'high', 'price'] to assertTimestamp is a bit odd

Reply:
yes, I am almost sure that we dont need that much props specifically to assertTimestamp, but it might need two (i.e. ` timestamp & datetime` fields), but what's important, the approach is good because its centralized and we dont need to maintain dozens of duplicate cases of `if (!('timestamp' in skippedProperties) assertTimestamp...` as we might be using `assertTimestamp` across 10+ test files. so, the centralized approach is quite much better, as we just use `assertTimestamp` in test files (passing with skipProps) and `if (!('timestamp' in skippedProperties)` line exists only once in that method instead of 10+ test files. also, in addition, we might easily change the logic from centralized place.